src/klement.jl

```diff
@@ -0,0 +1,62 @@
+"""
+```julia
+Klement()
+```
+
+A low-overhead implementation of [Klement](https://jatm.com.br/jatm/article/view/373).
+This method is non-allocating on scalar and static array problems.
+"""
+struct Klement <: AbstractSimpleNonlinearSolveAlgorithm end
+
+function SciMLBase.solve(prob::NonlinearProblem,
+                         alg::Klement, args...; abstol = nothing,
+                         reltol = nothing,
+                         maxiters = 1000, kwargs...)
+    f = Base.Fix2(prob.f, prob.p)
+    x = float(prob.u0)
+    fₙ = f(x)
+    T = eltype(x)
+    J = ArrayInterfaceCore.zeromatrix(x) + I
```

Comment:
This could be a fallback. If `x` is mutable, then we could do `J = ArrayInterfaceCore.zeromatrix(x); J[diagind(J)] .= one(eltype(x))`.

Reply:
Open an issue.